test/commands/git/sync_track_test.rb

```diff
@@ -79,17 +79,41 @@ class Git::SyncTrackTest < ActiveSupport::TestCase
 
     Git::SyncTrack.(track)
 
-    numbers = ::Track::Concept.find_by(uuid: 'd0fe01c7-d94b-4d6b-92a7-a0055c5704a3')
-    assert_includes track.concepts, numbers
+    assert track.concepts.find_by(uuid: 'd0fe01c7-d94b-4d6b-92a7-a0055c5704a3').present?
+  end
+
+  test "concept exercises use track concepts for taught concepts" do
+    csharp_track = create :track, slug: 'charp'
+    csharp_concept = create :track_concept, track: csharp_track, slug: 'basics'
+    fsharp_track = create :track, slug: 'fsharp', synced_to_git_sha: 'ab0b9be3162f6ec4ed6d7c46b55a8bf2bd117ffb'
+    fsharp_concept = create :track_concept, track: fsharp_track, slug: 'basics', uuid: 'f91b9627-803e-47fd-8bba-1a8f113b5215'
+
+    Git::SyncTrack.(fsharp_track)
+
+    fsharp_concept_exercise = fsharp_track.concept_exercises.find_by(uuid: '1fc8216e-6519-11ea-bc55-0242ac130003')
+    assert_includes fsharp_concept_exercise.taught_concepts, fsharp_concept
+    refute_includes fsharp_concept_exercise.taught_concepts, csharp_concept
+  end
+
+  test "concept exercises use track concepts for prerequisites" do
+    csharp_track = create :track, slug: 'charp'
+    csharp_concept = create :track_concept, track: csharp_track, slug: 'basics'
+    fsharp_track = create :track, slug: 'fsharp', synced_to_git_sha: 'ab0b9be3162f6ec4ed6d7c46b55a8bf2bd117ffb'
+    fsharp_concept = create :track_concept, track: fsharp_track, slug: 'basics', uuid: 'f91b9627-803e-47fd-8bba-1a8f113b5215'
+
+    Git::SyncTrack.(fsharp_track)
+
+    fsharp_concept_exercise = fsharp_track.concept_exercises.find_by(uuid: '9c2aad8a-53ee-11ea-8d77-2e728ce88125')
+    assert_includes fsharp_concept_exercise.prerequisites, fsharp_concept
+    refute_includes fsharp_concept_exercise.prerequisites, csharp_concept
   end
 
   test "adds new concept exercises defined in config.json" do
     track = create :track, slug: 'fsharp', synced_to_git_sha: 'ab0b9be3162f6ec4ed6d7c46b55a8bf2bd117ffb'
 
     Git::SyncTrack.(track)
 
-    cars_assemble = ConceptExercise.find_by(uuid: '6ea2765e-5885-11ea-82b4-0242ac130003')
-    assert_includes track.concept_exercises, cars_assemble
+    assert track.concept_exercises.find_by(uuid: '6ea2765e-5885-11ea-82b4-0242ac130003').present?
```

Comment:
```suggestion
    assert track.concept_exercises.where(uuid: '6ea2765e-5885-11ea-82b4-0242ac130003').exists?
```

Reply:
Exists simply checks for the presence in the db of something. `find_by` will init the object so is a lot heavier. It doesn't really matter for tests, but it's a good habit to use `exists?` if you're not then going to use the returned object.